profiling/src/profiling/mod.rs

```diff
@@ -790,14 +790,26 @@ impl Profiler {
 
                 let n_labels = labels.len();
 
+                let mut timestamp = NO_TIMESTAMP;
+                #[cfg(feature = "timeline")]
+                {
+                    let system_settings = self.system_settings.load(Ordering::SeqCst);
+                    // SAFETY: system settings are stable during a request.
+                    if unsafe { *ptr::addr_of!((*system_settings).profiling_timeline_enabled) } {
+                        if let Ok(now) = SystemTime::now().duration_since(UNIX_EPOCH) {
+                            timestamp = now.as_nanos() as i64;
+                        }
+                    }
```

Comment:
Do you think it's worth a helper routine for this? Or duplication is fine/better?

Reply:
I'm looking into this in another PR, good catch